integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -997,7 +997,7 @@ def test_count_distinct_with_nan_floats(data_gen):
 non_nan_all_basic_gens = [byte_gen, short_gen, int_gen, long_gen,
         # nans and -0.0 cannot work because of nan support in min/max, -0.0 == 0.0 in cudf for distinct and
         # Spark fixed ordering of 0.0 and -0.0 in Spark 3.1 in the ordering
-        FloatGen(no_nans=True, special_cases=[]), DoubleGen(no_nans=True, special_cases=[]),
```

Comment:
I do not know if this is the case, but if you remove `no_nans=True` e.g. always prefer the case with nans, do you _always_ get "a" nan? And does that mean that given a single batch do you not exercise the old path in `GpuMax` when nans are not detected? This is more of a question...

Reply:
Can we please put in a nan or two in the special case here? special_cases if it is set to None will add in Nans and other things as more frequently if it is set to [] it will not and then we rely on randomly generating a nan, which is not very common.